modules/aql/arrow-aql/src/main/kotlin/arrow/aql/Max.kt

```diff
@@ -0,0 +1,26 @@
+package arrow.aql
+
+import arrow.core.ForId
+import arrow.core.Id
+import arrow.core.identity
+import arrow.core.value
+import arrow.typeclasses.Foldable
+import kotlin.math.max
+
+interface Max<F> {
+
+  fun foldable(): Foldable<F>
+
+  infix fun <A, Z> Query<F, A, Z>.max(f: A.() -> Long): Query<ForId, Long, Long> =
```

Comment:
Instead of specializing for `Long`, require an `Order` instance: https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Order.kt

That way it works for everything and future-proofs it :D

Reply:
Make sense, I am gonna do some changes ;)